### ISSUE TITLE
RuntimeError in Kosmos-2 LoRa Fine-tuning

### ISSUE THREAD

**Issue by KevinHooah** [mit1280/fined-tuning#2]
Thank you for the great work! I am following the Kosmos-2 LoRa fine-tuning example, however, I get this error:

> RuntimeError: a leaf Variable that requires grad is being used in an in-place operation.

And the trace can be traced back to line 1147 in 'modeling_kosmos2.py': 

> inputs_embeds[img_input_mask.to(dtype=torch.bool)] = image_embeds.to(inputs_embeds.device).view(
>                 -1, image_embeds.size(-1)
>             )

Do you have any idea? Thanks!

**Comment by mit1280** [comment]
You can follow https://discuss.huggingface.co/t/kosmos-2-fine-tuning/75691/26 and re-start your kernel. 
Let me know if you still find any issue.

**Comment by KevinHooah** [comment]
Thanks for pointing out the solution. Issue fixed, in the Transformer 4.41.2, there are newer pieces of codes and the solution should be:

```
# add this line 
inputs_embeds = inputs_embeds.to(image_embeds.dtype)

# rest of the code:
if image_embeds is not None:
            inputs_embeds[img_input_mask.to(dtype=torch.bool)] = image_embeds.to(inputs_embeds.device).view(
                -1, image_embeds.size(-1)
            )

inputs_embeds = inputs_embeds * self.embed_scale
# ...
```
